tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+            seed=seed,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert task.max_trials == max_trials
+        assert task.seed == seed
+        assert task.task_id == task_id
+
+        # Directory should have one file (the trained model).
+        assert len(list(checkpoint_dir.iterdir())) == 1
+
+    def test_configuration(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Test that tasks have a proper configuration and that they can be created from it.
+        """
+        task_id = 0
+        kwargs = dict(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        first_task = self.Task(
+            **kwargs
+        )
+        configuration = first_task.configuration
+        assert len(configuration.keys()) == 1
+        key, config_dict = configuration.popitem()
+        assert key == self.Task.__qualname__
+        
+        for key, value in kwargs.items():
+            assert key in config_dict
+            assert config_dict[key] == value
+
+        # NOTE: This 'name' is the qualname of the class, which is used by the Factory metaclass'
+        # `__call__` method.
+        name = self.Task.__qualname__
+        from orion.benchmark.benchmark_client import _get_task
+        second_task = _get_task(name=name, **config_dict)
+        assert second_task.configuration == first_task.configuration        
+        
+
+    @pytest.mark.xfail(
+        reason="TODO: Not sure how to modify the `space.sample` for a particular space instance "
+        "without changing the 'global' RNG."
+    )
+    def test_space_sample_is_reproducible(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        task = self.Task(
+            max_trials=10,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        points = task._space.sample(10)
+        assert len(set(points)) == 10
+
+        # Create a new task:
+        second_task = self.Task(
+            max_trials=10,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert second_task._space.sample(10) == points
+    
+    def test_call_is_reproducible(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Two tasks created with the same args, given the same point, should produce the same
```

Comment:
As a sanity check, you should also verify that two tasks with different args give different points.

Reply:
Just a different seed would be enough for the sanity check.

Reply:
Added a test for that.